internal/mithril-persistence/src/database/repository/cardano_transaction_repository.rs

```diff
@@ -241,6 +241,20 @@ impl CardanoTransactionRepository {
         interval.to_range()
     }
 
+    /// Get the block number for a given slot number
+    pub async fn get_block_number_by_slot_number(
+        &self,
+        slot_number: SlotNumber,
+    ) -> StdResult<BlockNumber> {
```

Comment:
You should return a `StdResult<Option<BlockNumber>>` instead (and test it in `repository_get_block_number_by_slot_number`) of failing as we do for equivalent methods that fetch one result in db.